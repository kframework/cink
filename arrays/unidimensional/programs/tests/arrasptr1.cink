int main() {
  int a[5], * p, i;
  cout << "i = ";
  cin >> i;
  p = &(a[i]);
  cout << "a[i] = ";
  cin >> *p;
  if (*p < 3)
    *(++ p) = 2;
  else
    *(++ p) = 4;
  return *(p);
}
