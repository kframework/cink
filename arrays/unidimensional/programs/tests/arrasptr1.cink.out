i = a[i] = <T>
    <in>
        ListItem(#buffer ( "" ))
        ListItem(#istream ( 0 ))
    </in>
    <out>
        ListItem(#ostream ( 1 ))
        ListItem(#buffer ( "" ))
    </out>
    <control>
        <k>
            .K
        </k>
        <return>
            4
        </return>
        <fstack>
            .List
        </fstack>
        <sideEffects>
            .Bag
        </sideEffects>
        <env>
            main |-> #symScalLoc(1)
        </env>
    </control>
    <genv>
        main |-> #symScalLoc(1)
    </genv>
    <store>
        seq-array ( 5 , #symLoc(2) ) |-> store ( (store ( (const-array ( 5
           , 0 )) , 2 , 5 )) , 3 , 4 )
        #symScalLoc(1) |-> lambda ( .Decls @ ((((((((((int (a [ 5 ])) ;) ((p 
          of (pointer to int) ;) ((int i) ;))) ((cout << "i = ") ;)) ((cin >> 
          i) ;)) ((p = (& (a [ i ]))) ;)) ((cout << "a[i] = ") ;)) ((cin >> (*
           p)) ;)) (if ( ((* p) < 3) ) (((* (++ p)) = 2) ;) else (((* (++ p)) 
          = 4) ;))) (return (* p) ;)) )
        #symScalLoc(3) |-> select ( (seq-array ( 5 , #symLoc(2) )) , 3 )
        #symScalLoc(4) |-> 2
    </store>
</T>
